tool/site/server/src/routes/dispense/withdraw.ts

```diff
@@ -14,6 +20,22 @@ export default [
       const withdrawResponse = await epsClient.makeWithdrawRequest(withdrawRequest)
       const withdrawRequestHl7 = await epsClient.makeConvertRequest(withdrawRequest)
       const success = withdrawResponse.statusCode === 200
+      if (success) {
+        const prescriptionId = withdrawRequest.groupIdentifier?.value
+        const dispenseNotificationRequestKey = `dispense_notification_requests_${prescriptionId}`
+
+        const dispenseNotificationRequests = getSessionValue(dispenseNotificationRequestKey, request) as Bundle[]
+        const newDispenseNotifications = dispenseNotificationRequests.slice(0, -1)
```

Comment:
ive seen array.pop() used to remove the last element from the array. not sure its useful here since you want all elements except the last.

Reply:
can this ever be run on an empty array? (i think the -1 would throw?)

Reply:
yeah so array.pop() removes and returns the last element so could use it but I generally prefer not to mutate state so rather than mutating `dispenseNotifications` I declared a new const `newDispenseNotifications`. Not precious about this though so happy to change if preferred

I did wonder and was fairly sure this shouldn't run on and empty array but even if it does the `.slice(0, -1)` will just return an empty array which is probably the preferred behaviour